master/pyext/src/validation/__init__.py

```diff
@@ -12,527 +12,623 @@
-        lists=self.system.orphan_datasets
-        if len(lists)>0:
+        dataset_comp = {'ID': [], 'Dataset type': [],
+                        'Database name': [], 'Details': []}
+        lists = self.system.orphan_datasets
+        if len(lists) > 0:
             for i in lists:
                 try:
-                    loc=i.location.db_name
-                except AttributeError as error:
-                    loc=str('')
+                    loc = i.location.db_name
+                except AttributeError:
+                    loc = str('')
                 try:
-                    acc=i.location.access_code
-                except AttributeError as error:
-                    acc=str('Not listed')
+                    acc = i.location.access_code
+                except AttributeError:
+                    acc = str('Not listed')
                 dataset_comp['ID'].append(i._id)
-                if i.data_type=='unspecified' and i.details is not None:
+                if i.data_type == 'unspecified' and i.details is not None:
                     dataset_comp['Dataset type'].append(i.details)
                 else:
                     dataset_comp['Dataset type'].append(i.data_type)
                 dataset_comp['Database name'].append(str(loc))
                 if 'ComparativeModel' in str(i.__class__.__name__):
-                    acc1='template PDB ID: '+ acc
+                    acc1 = 'template PDB ID: ' + acc
                     dataset_comp['Details'].append(acc1)
                 elif 'PDB' in str(i.__class__.__name__):
-                    acc1='PDB ID: '+ acc
+                    acc1 = 'PDB ID: ' + acc
                     dataset_comp['Details'].append(acc1)
                 elif 'CX' in str(i.__class__.__name__):
-                    acc1=self.get_dataset_xl_info(i._id)
+                    acc1 = self.get_dataset_xl_info(i._id)
                     dataset_comp['Details'].append(acc1)
                 elif 'EM' in str(i.__class__.__name__):
-                    acc1='EMDB ID: '+acc
+                    acc1 = 'EMDB ID: '+acc
                     dataset_comp['Details'].append(acc1)
                 else:
                     dataset_comp['Details'].append(acc)
-                    
+
         return dataset_comp
-    
-    def get_atomic_coverage(self)->str:
+
+    def get_atomic_coverage(self) -> str:
         """Measure amount of atomic residues"""
         for _ in self.system.orphan_representations:
-            if self.check_sphere()==1:
-                flex=sum([int(x.asym_unit.seq_id_range[1])-int(x.asym_unit.seq_id_range[0])+1 for x in _ if not x.rigid])
-                rigid=sum([int(x.asym_unit.seq_id_range[1])-int(x.asym_unit.seq_id_range[0])+1 for x in _ if x.rigid])
-                percentage=str(round(rigid/(rigid+flex)*100))+'%'
+            if self.check_sphere() == 1:
+                flex = sum([int(x.asym_unit.seq_id_range[1]) -
```

Comment:
`seq_id_range` *should* always be an int (or `None`). If it isn't, I need to fix this in python-ihm.